pkg/kubelet/container/runtime.go

```diff
@@ -108,6 +108,11 @@ type Runtime interface {
 	// TODO: Change ContainerID to a Pod ID since the namespace is shared
 	// by all containers in the pod.
 	GetNetNS(containerID ContainerID) (string, error)
+	// Returns the container ID that represents the Pod, as passed to network
+	// plugins. For example if the runtime uses an infra container, returns
+	// the infra container's ContainerID.
+	// TODO: Change ContainerID to a Pod ID, see GetNetNS()
+	GetPodContainerID(*Pod) (*ContainerID, error)
```

Comment:
> Why are we assuming that a container will always be associated with a Pod? AFAIK, the network namespace is what is required for CNI right? If yes, why can't we have runtimes return the network namespace as part of Pod Status?

@vishh Yes, CNI requires the network namespace at this time.  But I don't think that's related at all to this PR, which only adds a way to get the mapping from Pod -> ContainerID, which is what is passed into the network plugins for SetUpPod/TearDownPod to uniquely identify a running pod.

WRT the larger question about returning network namespaces from setup, that's been discussed a bit in the context of the rkt runtime.  I don't personally think kubelet should have *anything* to do with network namespaces since all it really cares about is the IP address(es) of the pod.  Namespaces should be the domain of the network plugin.  But we have a few steps to get there...